sympy/physics/mechanics/functions.py

```diff
@@ -532,7 +539,7 @@ def _recurser(expr, sub_dict):
 
 def _fraction_decomp(expr):
     """Return num, den such that expr = num/den"""
-    if not isinstance(expr, Mul):
+    if not isinstance(expr, MulClass):
```

Comment:
In Cython `__new__` doesn't work.
```Cython
Error compiling Cython file:
------------------------------------------------------------
...
        import sage.all as sage
        return self.real_part()._sage_() + sage.I * self.imaginary_part()._sage_()

cdef class Add(Basic):

    def __new__(cls, *args):
   ^
------------------------------------------------------------

symengine_wrapper.pyx:964:4: __new__ method of extension type will change semantics in a future version of Pyrex and Cython. Use __cinit__ instead.
```

Also when using ``__cinit__`` you cannot return a different type
```Cython
Error compiling Cython file:
------------------------------------------------------------
...
        return self.real_part()._sage_() + sage.I * self.imaginary_part()._sage_()

cdef class Add(Basic):

    def __cinit__(cls, *args):
        return add(args)
                 ^
------------------------------------------------------------

symengine_wrapper.pyx:965:18: Return with value in void function
```